src/python/pants/reporting/reporting.py

```diff
@@ -88,9 +98,24 @@ def initialize(self, run_tracker, all_options, start_time=None):
 
     # Set up Zipkin reporting.
     zipkin_endpoint = self.get_options().zipkin_endpoint
+    trace_id = self.get_options().zipkin_trace_id
+    parent_id = self.get_options().zipkin_parent_id
+    if zipkin_endpoint is not None:
+      if trace_id is None and parent_id is not None or parent_id is None and trace_id is not None:
```

Comment:
`(trace_id is None) != (parent_id is None)` or `not (trace_id is None ^ parent_id is None)` would be more concise here :) something like:

```
if zipkin_endpoint is None and trace_id is not None and parent_id is not None:
  raise must set endpoint if trace and parent set
if (trace_id is None) != (parent_id is None):
  raise must either set both or neither
```